sklearnex/preview/decomposition/incremental_pca.py

```diff
@@ -0,0 +1,227 @@
+
+        if not hasattr(self, "n_samples_seen_"):
+            self.n_samples_seen_ = n_samples
+        else:
+            self.n_samples_seen_ += n_samples
+
+        onedal_params = {"n_components": self.n_components_, "whiten": self.whiten}
+
+        if not hasattr(self, "_onedal_estimator"):
+            self._onedal_estimator = self._onedal_incremental_pca(**onedal_params)
+        self._onedal_estimator.partial_fit(X, queue)
+        self._need_to_finalize = True
+
+    def _onedal_finalize_fit(self):
+        assert hasattr(self, "_onedal_estimator")
+        self._onedal_estimator.finalize_fit()
+        self._need_to_finalize = False
+
+    def _onedal_fit(self, X, queue=None):
+        if sklearn_check_version("1.2"):
+            self._validate_params()
+
+        if sklearn_check_version("1.0"):
+            X = self._validate_data(X, dtype=[np.float64, np.float32], copy=self.copy)
+        else:
+            X = check_array(
+                X,
+                dtype=[np.float64, np.float32],
+                copy=self.copy,
+            )
+
+        n_samples, n_features = X.shape
+
+        if self.batch_size is None:
+            self.batch_size_ = 5 * n_features
+        else:
+            self.batch_size_ = self.batch_size
+
+        self.n_samples_seen_ = 0
+        if hasattr(self, "_onedal_estimator"):
+            self._onedal_estimator._reset()
+
+        for batch in gen_batches(n_samples, self.batch_size_):
+            X_batch = X[batch]
+            self._onedal_partial_fit(X_batch, queue=queue)
+
+        self._onedal_finalize_fit()
+
+        return self
+
+    def _onedal_supported(self, method_name, *data):
+        patching_status = PatchingConditionsChain(
+            f"sklearn.decomposition.{self.__class__.__name__}.{method_name}"
+        )
+        return patching_status
+
+    _onedal_cpu_supported = _onedal_supported
+    _onedal_gpu_supported = _onedal_supported
+
+    def __getattr__(self, attr):
+        if attr in self._need_to_finalize_attrs:
+            if hasattr(self, "_onedal_estimator"):
+                if self._need_to_finalize:
+                    self._onedal_finalize_fit()
+                return getattr(self._onedal_estimator, attr)
+            else:
+                raise AttributeError(
+                    f"'{self.__class__.__name__}' object has no attribute '{attr}'"
+                )
+        if attr in self.__dict__:
+            return self.__dict__[attr]
+
+        raise AttributeError(
+            f"'{self.__class__.__name__}' object has no attribute '{attr}'"
+        )
+
+    def partial_fit(self, X, y=None, check_input=True):
+        dispatch(
+            self,
+            "partial_fit",
+            {
+                "onedal": self.__class__._onedal_partial_fit,
+                "sklearn": sklearn_IncrementalPCA.partial_fit,
+            },
+            X,
+            check_input=check_input,
+        )
+        return self
+
```

Comment:
is there a reason partial_fit and fit dont have @wrap_output_data but transform and fit_transform do?

Reply:
yes, it is. fit and partial fit doesn't return any data except for the estimator itself. but transform and fit_transform do. and wrap_output_data converts returned data to dpctl/dpnp array if input data had corresponding type.